### ISSUE TITLE
Sys - Updating dependencies

### ISSUE THREAD

**Issue by LuchoTurtle** [visual-space/visual-editor#236]
Hey!
Again, thanks for updating the `Flutter` SDK dependency in #234. I'm trying to get it working on a sample application but I'm getting a clash with dependencies. 

```
Because no versions of mockito match >5.4.2 <6.0.0 and mockito 5.4.2 depends on analyzer ^5.11.0, mockito ^5.4.2 requires analyzer ^5.11.0.
And because every version of visual_editor from git depends on i18n_extension ^4.2.1 which depends on analyzer ^3.0.0, mockito ^5.4.2 is incompatible with visual_editor from git.
So, because app depends on both visual_editor from git and mockito ^5.4.2, version solving failed.
```

This is because `visual_editor` is using some dependencies that are outdated. I figured that might as well update them! In my case, only (at least so far) `i18n_extensions` is complaining but we might as well update other dependencies, as long as they don't have meaningful breaking changes. 

I can open a PR :)

By the way, I'm using the latest `stable` version of Flutter.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.3, on macOS 13.4.1 22F770820d darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.82.0)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

Just an FIY, in the *next* version of `Flutter` (3.14.x), you will probably have to override three functions from [`TextSelectionDelegate`](https://api.flutter.dev/flutter/services/TextSelectionDelegate-mixin.html) 😉.

**Comment by adrian-moisa** [comment]
I will highly appreciate if you manage to upgrade them. Let me know when the PR is ready. And we will merge it. Also I haven't had time to double check on the tests PR. I will get to that one as well, wanted to reserve some time to have a quick chat with you on discord to figure out what's the plan there. Maybe tomorrow or Thursday Cheers!

**Comment by LuchoTurtle** [comment]
The PR is practically done. I will open it after I get through a medical appointment I'm currently waiting on.

Speaking of which, in my closed PR #235, I underwent the replacement of `gallery_saver` to `gal`. See https://github.com/CarnegieTechnologies/gallery_saver/issues/206#issue-1855285526.

I did this because `gallery_saver` has not been maintained for 2 years and because it was dependent on an outdated version `http` (if you see the opened issues in `gallery_saver`, many people complain about this).

I don't know if this would be something you'd be interested in, but I just wanted to point it out 👍

**Comment by adrian-moisa** [comment]
update in a different ticket
